07_errors/hsy3418/store_test.go

```diff
@@ -0,0 +1,131 @@
+package main
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/suite"
+)
+
+func (suite *storerTestSuite) SetupTest() {
+	suite.store = suite.makeStore()
+	suite.toBeCreatedPuppy = Puppy{ID: 101, Breed: "Poodle", Colour: "White", Value: 1000.5}
+	suite.existsPuppy = Puppy{ID: 102, Breed: "Poodle", Colour: "White", Value: 1280.5}
+	suite.toBeUpdatedPuppy = Puppy{ID: 102, Breed: "Poodle", Colour: "White", Value: 2000}
+	suite.invalidPuppy = Puppy{ID: 103, Breed: "Poodle", Colour: "White", Value: -1000}
+	suite.invaildError = ErrorEf(ErrInvalidInput, "The puppy value is invalidate")
+	suite.invalidIDError = ErrorEf(ErrNotFound, "This puppy does not exist")
+	suite.dupicatesError = ErrorEf(ErrDuplicate, "This puppy exists ")
+	err := suite.store.CreatePuppy(suite.existsPuppy)
+	if err != nil {
+		suite.FailNow("Failed to setup test")
+	}
+}
+
+type storerTestSuite struct {
+	suite.Suite
+	store            Storer
+	makeStore        func() Storer
+	toBeCreatedPuppy Puppy
+	existsPuppy      Puppy
+	toBeUpdatedPuppy Puppy
+	invalidPuppy     Puppy
+	invaildError     error
+	invalidIDError   error
+	dupicatesError   error
+}
+
+//successfully create puppy, new puppy, existing puppy, value <0
+func (suite *storerTestSuite) TestCreatePuppy() {
+	assert := assert.New(suite.T())
+	testCases := []struct {
+		title    string
+		input    Puppy
+		expected error
+	}{
+		{"Create new puppy", suite.toBeCreatedPuppy, nil},
+		{"Create existing puppy", suite.toBeCreatedPuppy, suite.dupicatesError},
+		{"Create a invalid puppy", suite.invalidPuppy, suite.invaildError},
+	}
+	for _, tc := range testCases {
+		tc := tc
+		suite.T().Run(tc.title, func(t *testing.T) {
+			err := suite.store.CreatePuppy(tc.input)
+			assert.Equal(tc.expected, err)
+		})
+	}
+
+}
+
+//
```

Comment:
Looks like you have gone to put a comment here and then forgotten. Test's don't really need documentation on their function names, so this can be removed.